### DISCUSSION TITLE
Homebrew installation on the new Mac(M1).

### DISCUSSION THREAD

**Discussion by AayushBajra12** [Homebrew/discussions#446] [Getting started]
I installed homebrew in my mac M1. It says installation successful but when I enter commands with brew, it gives brew command not found error. It was working before. I have been trying to install mongodb with brew and was getting errors so I uninstalled homebrew and got the command not found error after reinstalling it.
I tried installing it with Rosetta on and off.

**Comment by carlocab** [comment]
You need to add Homebrew to your `PATH`. Do

    echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> $HOME/.zprofile

then restart your terminal. This is actually what's suggested at the very end of the install script.

**Reply by AayushBajra12** [reply]
It gives me a command not found: brew message

**Reply by carlocab** [reply]
Sorry, there was a type in the first command I showed. I've edited it now. You can also edit your `.zprofile` in TextEdit by doing

    open $HOME/.zprofile

Make sure to remove any line that says `brew shellenv` from my first mistaken post.

**Reply by nchakaben** [reply]
Yeah, it works. After installing Homebrew 3.2.3 on Big Sur, this was already included with an error:

```
echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> /Users/[your-user-name]/.zprofile
eval "$(/opt/homebrew/bin/brew shellenv)"
```

It should have been:

```
echo 'eval "$(/opt/homebrew/bin/brew shellenv)"'
eval "$(/opt/homebrew/bin/brew shellenv)"
```

The` >> /Users/[your-user-name]/.zprofile `part keeps on re-adding the  echo command in the .zprofile every single time a terminal session is launch. Lucky me it was breaking anything. But I had to edit a long list of "eval" lines.

**Reply by gromgit** [reply]
Actually, there should only be _one_ Homebrew-related line in your `.zprofile`:
```
eval "$(/opt/homebrew/bin/brew shellenv)"
```
That's all. Unfortunately, the original instructions were somewhat confusing:
```
- Add Homebrew to your **PATH** in `/path/to/profile`:
```
so after the 3.2.3 release, it was changed to the explicit:
```
- Run these two commands in your terminal to add Homebrew to your **PATH**:
```

**Reply by nchakaben** [reply]
@gromgit Yeah, you're right. It should be that single line.

**Reply by liomoa** [reply]
It worked for me. M2 Mac mini Ventura. Thanks!

**Reply by belblin** [reply]
Single line worked for me! Thanks
`eval "$(/opt/homebrew/bin/brew shellenv)"`

**Reply by Adrian-Samoticha** [reply]
Adding `echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> $HOME/.zprofile` to `.zprofile` will cause `eval "$(/opt/homebrew/bin/brew shellenv)"` to be added to `.zprofile` **every time zsh is launched.**

That means that any time zsh is launched, its launch time will become slightly longer. After ~2000 launches it took more than 10 seconds on my machine. Please only add `eval "$(/opt/homebrew/bin/brew shellenv)"` to `.zprofile`, instead.

**Reply by bobgt** [reply]
Great! The single line command worked for me! Thanks!

**Reply by kuangkaocompassion** [reply]
Thanks a lot!!!!

**Reply by josefdclxvi** [reply]
this works for me in sonomaOS, thank you so much

**Reply by dong4j** [reply]
@Adrian-Samoticha Thank you for your help, I have cleaned up over 4000+ identical names in the .zprofile file.

**Comment by vinilet** [comment]
How can undelete this command? Cause I have uninstalled Homebrew and It keep on appear this warning when I open a new terminal

/Users/<user>/.zprofile:1: no such file or directory: /opt/homebrew/bin/brew

Thanks

**Reply by gromgit** [reply]
From a Terminal session:

1. `open ~/.zprofile`
2. Delete every line that mentions `/opt/homebrew/bin/brew`
3. Save the file

**Reply by vinilet** [reply]
Thanks It works!

**Reply by jasonshu1229** [reply]
==> Checking for `sudo` access (which may request your password)...
eval $(/opt/homebrew/bin/brew shellenv)
eval $(/opt/homebrew/bin/brew shellenv)
eval $(/opt/homebrew/bin/brew shellenv)

**Comment by amidpro** [comment]
Eventually understood the instructions and successfully downloaded brew, thank you all!

Just wanted to clearly state the solution:

COMPLETELY IGNORE "NEXT STEPS" SECTION COMMAND! COPY SOMEWHERE EXACT CODE SHOWN BELOW, CHANGE "YOURUSERNAME" - REMOVE AND REPLACE WITH YOUR MAC USERNAME:

echo 'eval "$(/opt/homebrew/bin/brew)"' >> /Users/YOURUSERNAME/.zprofile
eval "$(/opt/homebrew/bin/brew)"

SHOULD SHOW EXAMPLE USAGE: BREW COMMANDS, ie, brew search, brew info etc.

TO DOUBLE-CHECK BREW INSTALLED:
brew -v

WILL RETURN VERSION INSTALLED

**Reply by gromgit** [reply]
> COMPLETELY IGNORE "NEXT STEPS" SECTION COMMAND!

And then you post _exactly the same commands_ as the **Next steps** section would have, but your first command only works if the user is running Zsh. The Homebrew installer, on the other hand, actually _checks what shell the user is running_, and customizes the first command accordingly.

I'd say instead: _Read the **Next steps** section carefully, and do exactly what it says._

**Reply by prathameshshirole** [reply]
worked thanks

**Comment by dhruvsrivastava** [comment]
The solution does not work for me.

When I open a new terminal, I need to re-run the command every time.

```
echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> $HOME/.zprofile
```

I updated my $HOME/.zprofile with the command but still I need to re-run this command when I start a new terminal.

Any ideas?

**Reply by gromgit** [reply]
Have you tried logging out, then logging back in again?

You should also take a look at your `$HOME/.zprofile`. If you did exactly what you said:
> I need to re-run the command every time

your `.zprofile` is probably filled with _many_ copies of the same `eval ...` line. You only need one copy of that command.

**Reply by shiroamada** [reply]
> Have you tried logging out, then logging back in again?
> 
> You should also take a look at your `$HOME/.zprofile`. If you did exactly what you said:
> 
> > I need to re-run the command every time
> 
> your `.zprofile` is probably filled with _many_ copies of the same `eval ...` line. You only need one copy of that command.

same issue for me, I am sure the .zprofile is running, because I `echo "Loading .zprofile"`, when I start a new tab, this line is printed. but still brew is not working. Even I added to `.zshrc` also same thing it will not run in iterm2.

**Reply by gromgit** [reply]
@shiroamada Just to be sure, what's the output of `echo $SHELL`?

**Reply by shiroamada** [reply]
the output is `/bin/zsh`

**Reply by shiroamada** [reply]
https://user-images.githubusercontent.com/2062946/218297423-be65ff45-4c91-4f86-b343-b58816e1a035.mp4

**Reply by gromgit** [reply]
What's the output of `zsh -xlc true | grep -w PATH`?

**Reply by shiroamada** [reply]
```
» zsh -xlc true | grep -w PATH                                                                          
+/etc/zprofile:6> [ -x /usr/libexec/path_helper ']'
+/etc/zprofile:7> /usr/libexec/path_helper -s
+/etc/zprofile:7> eval 'PATH="/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin";' export 'PATH;' 'MANPATH="/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:";' export 'MANPATH;'
+(eval):1> PATH='/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+(eval):1> export PATH
+(eval):1> MANPATH=/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:
+(eval):1> export MANPATH
+/Users/ken/.zprofile:2> /opt/homebrew/bin/brew shellenv
+/Users/ken/.zprofile:2> eval $'export HOMEBREW_PREFIX="/opt/homebrew";\nexport HOMEBREW_CELLAR="/opt/homebrew/Cellar";\nexport HOMEBREW_REPOSITORY="/opt/homebrew";\nexport PATH="/opt/homebrew/bin:/opt/homebrew/sbin${PATH+:$PATH}";\nexport MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";\nexport INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";'
+(eval):1> export HOMEBREW_PREFIX=/opt/homebrew
+(eval):2> export HOMEBREW_CELLAR=/opt/homebrew/Cellar
+(eval):3> export HOMEBREW_REPOSITORY=/opt/homebrew
+(eval):4> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+(eval):5> export MANPATH=/opt/homebrew/share/man:/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man::
+(eval):6> export INFOPATH=/opt/homebrew/share/info:/opt/homebrew/share/info:/opt/homebrew/share/info:
+/Users/ken/.zprofile:3> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+/Users/ken/.zprofile:4> echo 'Loading .zprofile'
+zsh:1> true
```

**Reply by gromgit** [reply]
Whoops, forgot that this command line invokes a command directly, so it doesn't source `~/.zshrc`.

Try just running `zsh -xl` and posting the output.

**Reply by shiroamada** [reply]
```
» zsh -xl                                                                                   
+/etc/zprofile:6> [ -x /usr/libexec/path_helper ']'
+/etc/zprofile:7> /usr/libexec/path_helper -s
+/etc/zprofile:7> eval 'PATH="/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin";' export 'PATH;' 'MANPATH="/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:";' export 'MANPATH;'
+(eval):1> PATH='/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+(eval):1> export PATH
+(eval):1> MANPATH=/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:
+(eval):1> export MANPATH
+/Users/ken/.zprofile:2> /opt/homebrew/bin/brew shellenv
+/Users/ken/.zprofile:2> eval $'export HOMEBREW_PREFIX="/opt/homebrew";\nexport HOMEBREW_CELLAR="/opt/homebrew/Cellar";\nexport HOMEBREW_REPOSITORY="/opt/homebrew";\nexport PATH="/opt/homebrew/bin:/opt/homebrew/sbin${PATH+:$PATH}";\nexport MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";\nexport INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";'
...
+/Users/ken/.oh-my-zsh/custom/themes/agnoster.zsh-theme:277> RPROMPT='$(my_right_prompt)'
+/Users/ken/.oh-my-zsh/custom/themes/agnoster.zsh-theme:278> PROMPT='%{%f%b%k%}$(build_prompt) '
+/Users/ken/.zshrc:113> export PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin
+/Users/ken/.zshrc:136> export PATH='~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
+/Users/ken/.zshrc:139> alias 'vm=ssh vagrant@127.0.0.1 -p 2222'
+/Users/ken/.zshrc:142> alias 'vup=cd ~/Homestead/ && vagrant up'
+/Users/ken/.zshrc:145> alias 'vs=cd ~/Homestead/ && vagrant suspend'
+/Users/ken/.zshrc:148> export LC_ALL=en_US.UTF-8
+/Users/ken/.zshrc:149> export LANG=en_US.UTF-8
```

**Reply by gromgit** [reply]
After dumping your debug output to `output.txt`, the problem becomes clear:
```shell
$ grep PATH= output.txt 
+/etc/zprofile:7> eval 'PATH="/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin";' export 'PATH;' 'MANPATH="/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:";' export 'MANPATH;'
+(eval):1> PATH='/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+(eval):1> MANPATH=/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:
+/Users/ken/.zprofile:2> eval $'export HOMEBREW_PREFIX="/opt/homebrew";\nexport HOMEBREW_CELLAR="/opt/homebrew/Cellar";\nexport HOMEBREW_REPOSITORY="/opt/homebrew";\nexport PATH="/opt/homebrew/bin:/opt/homebrew/sbin${PATH+:$PATH}";\nexport MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";\nexport INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";'
+(eval):4> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+(eval):5> export MANPATH=/opt/homebrew/share/man:/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man::
+(eval):6> export INFOPATH=/opt/homebrew/share/info:/opt/homebrew/share/info:/opt/homebrew/share/info:
+/Users/ken/.zprofile:3> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+/Users/ken/.zshrc:2> export PATH='/Users/ken/bin:/opt/homebrew/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+/Users/ken/.zshrc:8> eval export 'HOMEBREW_PREFIX="/opt/homebrew";' export 'HOMEBREW_CELLAR="/opt/homebrew/Cellar";' export 'HOMEBREW_REPOSITORY="/opt/homebrew";' export 'PATH="/opt/homebrew/bin:/opt/homebrew/sbin${PATH+:$PATH}";' export 'MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";' export 'INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";'
+(eval):1> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/Users/ken/bin:/opt/homebrew/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
+(eval):1> export MANPATH=/opt/homebrew/share/man:/opt/homebrew/share/man:/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:::
+(eval):1> export INFOPATH=/opt/homebrew/share/info:/opt/homebrew/share/info:/opt/homebrew/share/info:/opt/homebrew/share/info:
+/Users/ken/.zshrc:113> export PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin
+/Users/ken/.zshrc:136> export PATH='~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
+/Users/ken/.zshrc:202> export PATH='/usr/local/sbin:~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
+/Users/ken/.zshrc:205> export PATH='/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
+/Users/ken/.zshrc:206> export PATH='/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
```
You're completely _resetting_ your `PATH` in line 113 of `~/.zshrc`, wiping out `shellenv`'s customizations.

**Reply by shiroamada** [reply]
> After dumping your debug output to `output.txt`, the problem becomes clear:
> 
> ```shell
> $ grep PATH= output.txt 
> +/etc/zprofile:7> eval 'PATH="/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin";' export 'PATH;' 'MANPATH="/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:";' export 'MANPATH;'
> +(eval):1> PATH='/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
> +(eval):1> MANPATH=/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:
> +/Users/ken/.zprofile:2> eval $'export HOMEBREW_PREFIX="/opt/homebrew";\nexport HOMEBREW_CELLAR="/opt/homebrew/Cellar";\nexport HOMEBREW_REPOSITORY="/opt/homebrew";\nexport PATH="/opt/homebrew/bin:/opt/homebrew/sbin${PATH+:$PATH}";\nexport MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";\nexport INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";'
> +(eval):4> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
> +(eval):5> export MANPATH=/opt/homebrew/share/man:/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man::
> +(eval):6> export INFOPATH=/opt/homebrew/share/info:/opt/homebrew/share/info:/opt/homebrew/share/info:
> +/Users/ken/.zprofile:3> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
> +/Users/ken/.zshrc:2> export PATH='/Users/ken/bin:/opt/homebrew/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
> +/Users/ken/.zshrc:8> eval export 'HOMEBREW_PREFIX="/opt/homebrew";' export 'HOMEBREW_CELLAR="/opt/homebrew/Cellar";' export 'HOMEBREW_REPOSITORY="/opt/homebrew";' export 'PATH="/opt/homebrew/bin:/opt/homebrew/sbin${PATH+:$PATH}";' export 'MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";' export 'INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";'
> +(eval):1> export PATH='/opt/homebrew/bin:/opt/homebrew/sbin:/Users/ken/bin:/opt/homebrew/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/Users/ken/.composer/vendor/bin:/Users/ken/.composer/vendor/bin'
> +(eval):1> export MANPATH=/opt/homebrew/share/man:/opt/homebrew/share/man:/usr/share/man:/usr/local/share/man:/opt/homebrew/share/man:::
> +(eval):1> export INFOPATH=/opt/homebrew/share/info:/opt/homebrew/share/info:/opt/homebrew/share/info:/opt/homebrew/share/info:
> +/Users/ken/.zshrc:113> export PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin
> +/Users/ken/.zshrc:136> export PATH='~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
> +/Users/ken/.zshrc:202> export PATH='/usr/local/sbin:~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
> +/Users/ken/.zshrc:205> export PATH='/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
> +/Users/ken/.zshrc:206> export PATH='/opt/homebrew/opt/php@8.2/sbin:/opt/homebrew/opt/php@8.2/bin:/usr/local/sbin:~/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/ken/.composer/vendor/bin'
> ```
> 
> You're completely _resetting_ your `PATH` in line 113 of `~/.zshrc`, wiping out `shellenv`'s customizations.

Thanks for figure it out, the ~/.zshrc was bring from old Mac, I should be careful and checking for this.
Thanks for helping @gromgit

**Reply by EFH500** [reply]
I am a complete newbie (a 57 yo lawyer/grandmother trying get through the steps to instal Automatic 1111) and I just followed all of these instructions and I was able to fix the issue and install homebrew.  I have no idea what or how I did it, or what your instructions mean, but I thank you all from the bottom of my heart. I had to sign up to Github just to express my appreciation.

**Comment by froodntoodn** [comment]
After all this back and forth I have to say I'm quite confused now. Can I just get a final version of what the command should be? I mean just spelled out without take-backs? I'm scared of messing things up on my machine if I type it wrong.
I run Big Sur on a mac mini with M1 chip. Thanks!

**Reply by gromgit** [reply]
The installer command you want is [front-and-center on the Homebrew home page](https://brew.sh/):
```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
"All this back and forth" is what happens when you don't do ***EXACTLY*** what the installer tells you to do under *Next Steps*. Common mistakes include:
1. Ignoring the exact commands under *Next Steps*, and trying to do something different that you think gets you the same result.
2. Manually retyping the commands you're given...and introducing typos.

The procedure most users have encountered zero problems with:
1. Selecting the command text with your mouse.
2. Select your terminal's "Copy" function.
3. Select your terminal's "Paste" function.
4. Hit the `Enter` key (just in case your paste leaves the last command "hanging").

**Comment by viviboxwerbung** [comment]
brew --version
